subsys/bluetooth/host/l2cap.c

```diff
@@ -3035,6 +3036,105 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 	return 0;
 }
 
+#if defined(CONFIG_BT_L2CAP_RECONFIGURE_EXPLICIT)
+int bt_l2cap_ecred_chan_reconfigure_explicit(struct bt_l2cap_chan **chans, uint16_t mtu,
+					     uint16_t mps)
+{
+	struct bt_l2cap_ecred_reconf_req *req;
+	struct bt_conn *conn = NULL;
+	struct bt_l2cap_le_chan *ch;
+	struct net_buf *buf;
+	bool multiple_chan;
+	uint8_t ident;
+	int i;
+
+	LOG_DBG("chans %p mtu 0x%04x", chans, mtu);
+
+	if (!chans) {
+		return -EINVAL;
+	}
+
+	if (chans[0] == NULL) {
+		return -EINVAL;
+	}
+
+	if (mps > BT_L2CAP_RX_MTU) {
+		return -EINVAL;
+	}
+
+	multiple_chan = chans[1] != NULL;
+
+	for (i = 0; i < L2CAP_ECRED_CHAN_MAX_PER_REQ; i++) {
+		if (!chans[i]) {
+			break;
+		}
+
+		/* validate that all channels are from same connection */
+		if (conn) {
+			if (conn != chans[i]->conn) {
+				return -EINVAL;
+			}
+		} else {
+			conn = chans[i]->conn;
+		}
+
+		/* validate MTU is not decreased */
+		if (mtu < BT_L2CAP_LE_CHAN(chans[i])->rx.mtu) {
+			return -EINVAL;
+		}
+
+		/* MPS is not allowed to decrease when reconfiguring multiple channels.
+		 * Core Specification 3.A.4.27 v6.0
+		 */
+		if (multiple_chan && mps < BT_L2CAP_LE_CHAN(chans[i])->rx.mps) {
+			return -EINVAL;
+		}
```

Comment:
If you do 
```suggestion
		if (i > 0 && mps < BT_L2CAP_LE_CHAN(chans[i])->rx.mps) {
			return -EINVAL;
		}
```

Then you can omit the `multiple_chan` variable. It is also OK as is, as that is quite explicit and easy to read

Reply:
I'll keep it as is then. I clarified it like this while explaining the code to someone, so I think it is good to have.